File: packages/cloud-adapters/database/aws.go
package database

import (
	"context"
	"errors"
)

// AwsAdapter is the RDS-backed database adapter. Stubbed pending the AWS-cloud milestone.
type AwsAdapter struct{}

func NewAwsAdapter() *AwsAdapter { return &AwsAdapter{} }

func (*AwsAdapter) ProvisionInstance(context.Context, string, string) (Handle, error) {
	return Handle{}, errors.New("NotImplemented: cloud-adapters/database/aws.ProvisionInstance — RDS impl lands in the AWS-cloud milestone")
}
func (*AwsAdapter) DescribeInstance(context.Context, string) (Handle, error) {
	return Handle{}, errors.New("NotImplemented: cloud-adapters/database/aws.DescribeInstance — RDS impl lands in the AWS-cloud milestone")
}
func (*AwsAdapter) DeleteInstance(context.Context, string) error {
	return errors.New("NotImplemented: cloud-adapters/database/aws.DeleteInstance — RDS impl lands in the AWS-cloud milestone")
}

File: packages/cloud-adapters/database/aws.ts
import type { DatabaseAdapter } from "./interface";

const notImpl = (op: string): never => {
  throw new Error(`NotImplemented: cloud-adapters/database/aws.${op} — RDS impl lands in the AWS-cloud milestone`);
};

export const aws: DatabaseAdapter = {
  provisionInstance: async () => notImpl("provisionInstance"),
  describeInstance:  async () => notImpl("describeInstance"),
  deleteInstance:    async () => notImpl("deleteInstance"),
};
